tendrl/gluster_integration/sds_sync/brick_device_details.py

```diff
@@ -0,0 +1,92 @@
+from tendrl.commons.event import Event
+from tendrl.commons.message import Message
+from tendrl.commons.utils import cmd_utils
+
+
+def get_brick_source_and_mount(brick_path):
+    # source and target correspond to fields "Filesystem" and
+    # "Mounted on" from df command output. The below command
+    # gives the filesystem and mount point for a given path,
+    # Eg: "/dev/mapper/tendrlMyBrick4_vg-tendrlMyBrick4_lv " \
+    #     "/tendrl_gluster_bricks/MyBrick4_mount"
+
+    command = "df --output=source,target " + brick_path.split(":")[-1]
+    cmd = cmd_utils.Command(command)
+    out, err, rc = cmd.run()
+    if rc != 0:
+        Event(
+            Message(
+                priority="error",
+                publisher=NS.publisher_id,
+                payload={
+                    "message": "%s command failed: %s" % (
+                        command, err
+                    )
+                }
+            )
+        )
+        return None, None
+    return out.split("\n")[-1].split()
+
+
+def get_disks(dev, disks):
+    # recursively find the physical devices under the
+    # given device
+    if dev.type == "disk":
+        if dev.path not in disks:
+            disks.append(dev.path)
+        return
+
+    for d in dev.ancestors:
+        get_disks(d, disks)
+
+    return
+
+
+def update_brick_device_details(brick_name, brick_path, devicetree):
+    mount_source, mount_point = get_brick_source_and_mount(brick_path)
+    if not mount_source or not mount_point:
+        Event(
+            Message(
+                priority="error",
+                publisher=NS.publisher_id,
+                payload={
+                    "message": "Cound not update brick device details"
+                }
+            )
+        )
+        return
+
+    device = devicetree.resolveDevice(mount_source)
+    size = int(device.size.to_integral())
+    lv = None
+    pool = None
+    vg = None
+    pvs = None
+    disks = []
+
+    # get all the physical devices that is under this
+    # device. Tendrl must be having info about underlying
+    # physical devices of a gluster brick.
+    for d in device.disks:
+        get_disks(d, disks)
+
+    if device.type in ("lvmthinlv", "lvmlv"):
```

Comment:
In gluster the bricks are either lvm logical volumes (thinlv or normal) or partitions. So I added the above condition to make sure lvm related details (underlying pool, vg and pv) are assigned only if the brick is a lv or thin lv.
Do you see any issue here ?